src/main/java/com/nvidia/spark/rapids/jni/JSONUtils.java

```diff
@@ -166,80 +166,65 @@ public static ColumnVector extractRawMapFromJsonString(ColumnView input, JSONOpt
   }
 
   /**
-   * Extract key-value pairs for each output map from the given json strings. This method is
-   * similar to {@link #extractRawMapFromJsonString(ColumnView, JSONOptions)} but is deprecated.
-   *
-   * @deprecated This method is deprecated since it does not have parameters to control various
-   * JSON reader behaviors.
+   * Parse a JSON string into a struct column following by the given data schema.
    *
    * @param input The input strings column in which each row specifies a json object
-   * @return A map column (i.e., a column of type {@code List<Struct<String,String>>}) in
-   *         which the key-value pairs are extracted directly from the input json strings
+   * @param schema The schema of the output struct column
+   * @param opts The options for parsing JSON strings
```

Comment:
nit: Could we indicate that some of the options will be ignored and are hard coded to what Spark needs?